src/commondao/common.dao.ts

```diff
@@ -589,6 +601,7 @@ export class CommonDao<
     const dbm = await this.bmToDBM(bm, opt)
     const table = opt.table || this.cfg.table
     if (opt.ensureUniqueId && idWasGenerated) await this.ensureUniqueId(table, dbm)
+    else if (this.cfg.immutable === true && !idWasGenerated) await this.ensureUniqueId(table, dbm)
```

Comment:
My comment is not about this line, but in general, where in `save` method do you require object to be immutable?

As I understand, the only way to achieve it is to LOAD the object first (like the `patch` method does), but I don't see where it's done. Maybe I'm missing something.

UPD: ok, found it inside `ensureUniqueId` method!

UPD2: but in `ensureUniqueId` if it loads an existing item by id - it throws `NON_UNIQUE_ID` error, while you want it to throw your newly added error (and tests probably don't verify the exact error message)

Reply:
Indeed it felt a bit hacky to piggyback on the existing error message, I'll refactor.